### ISSUE TITLE
enable VPCSC dryrun mode 

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1209]
### TL;DR

Enable VPCSC dryrun mode to mitigate issues with flaky failures in CI tests.

### Expected behavior

[Integration tests](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/build/int.cloudbuild.yaml#L141) create a perimeter along with other resources, configure an access level exception to work with resources inside the perimeter, then eventually tear down the whole perimeter.

When done in the correct order and with the correct configurations, this should all work without errors.

### Observed behavior

There is a high rate of flaky CI failures related to VPCSC errors. 

VPCSC has a long propagation delay when modifying the perimeter.  Even after a command to remove the perimeter, subsequent steps like [`destroy-networks`](https://github.com/terraform-google-modules/terraform-example-foundation/blob/a759ee9ddb429b58b5750523922c07eb83fb152d/build/int.cloudbuild.yaml#L141) have a high failure rate with the error `SECURITY_POLICY_VIOLATED`

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
n/a
```

### Additional information

Regardless of the immediate issue with CI pipelines, the change to VPCSC perimeter design was already planned for a more significant design overhaul in a later version. This change aligns to existing best practices and the future planned design work.

Enabling a VPCSC perimeter in a dryrun mode is the recommended [best practice from product documentation](https://cloud.google.com/vpc-service-controls/docs/enable#dry-run) and also matches the [guidance in the Enterprise foundations blueprint concept guide](https://cloud.google.com/architecture/security-foundations/operation-best-practices#protect-resources) for how to safely adopt VPCSC.

**Comment by eeaton** [comment]
Addressed by PR 1185
